localstack/services/apigateway/helpers.py

```diff
@@ -1241,3 +1241,54 @@ def _params(name=None):
             input_ctx["params"] = _params
             data = aws_stack.render_velocity_template(template, input_ctx, variables=variables)
     return data
+
+
+def apply_integration_response_template(context: ApiInvocationContext):
```

Comment:
Yes, this is a combination of both actually because none of the two is sufficient to handle response templates, hence this new function that will be the "standard" to handle response templates, the other two might actually disappear if not used outside of this scope or refactored into functions shared across `request_templates` and `response_templates`. This is not an accident, it's done to avoid breaking existing usages of those functions.